components/fs-data-provider/src/bin/icu4x-cldr-export.rs

```diff
@@ -0,0 +1,192 @@
+use clap::{App, Arg, ArgGroup};
+use icu_cldr_json_data_provider::CldrJsonDataProvider;
+use icu_cldr_json_data_provider::CldrPaths;
+use icu_data_provider::icu_data_key;
+use icu_data_provider::iter::IterableDataProvider;
+use icu_fs_data_provider::export::fs_exporter;
+use icu_fs_data_provider::export::serializers;
+use icu_fs_data_provider::export::FilesystemExporter;
+use icu_fs_data_provider::manifest;
+use std::ffi::OsStr;
+use std::fmt;
+use std::path::PathBuf;
+
+// #[derive(Debug)]
+enum Error {
+    Unsupported(&'static str),
+    Export(icu_fs_data_provider::FsDataError),
+    DataProvider(icu_data_provider::DataError),
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::Unsupported(message) => write!(f, "Unsupported: {}", message),
+            Error::Export(error) => write!(f, "{}", error),
+            Error::DataProvider(error) => write!(f, "{}", error),
+        }
+    }
+}
+
+impl fmt::Debug for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        (self as &dyn fmt::Display).fmt(f)
+    }
+}
+
+impl From<icu_fs_data_provider::FsDataError> for Error {
+    fn from(err: icu_fs_data_provider::FsDataError) -> Error {
+        Error::Export(err)
+    }
+}
+
+impl From<icu_data_provider::DataError> for Error {
+    fn from(err: icu_data_provider::DataError) -> Error {
+        Error::DataProvider(err)
+    }
+}
+
+fn main() -> Result<(), Error> {
+    let matches = App::new("ICU4X Data Exporter")
+        .version("0.0.1")
+        .author("The ICU4X Project Developers")
+        .about("Export CLDR JSON into the ICU4X data schema")
+        .arg(
+            Arg::with_name("VERBOSE")
+                .short("v")
+                .long("verbose")
+                .help("Enable verbose logging."),
+        )
+        .arg(
+            Arg::with_name("DRY_RUN")
+                .short("n")
+                .long("dry-run")
+                .help("Do not touch the filesystem (consider using with -v)."),
+        )
+        .arg(
+            Arg::with_name("ALIASING")
```

Comment:
> What is "aliasing"?
> Does it have something to do with the fact that the same rules apply to several language?

Basically yes.  The "aliasing" setting is intended to be a toggle switch to let you choose how you want aliases persisted on the filesystem.  Right now there are two choices: no-aliases and symlink.  No-aliases means that all locales with data have full JSON files persisted onto the filesystem.  Symlink means that locales with identical data have filesystem symlinks to the source of truth.  The idea is that symlinks are lighterweight than full JSON files, which may or may not hold true for small data.

> Why don't we have JSON files (in this PR) for all the locales?

Because the data in this PR is fake data that I put together for testing purposes.  I intend to follow up with a real export in a future PR.  I also want to discuss whether we want the full data living in this repo or whether it should live in its own repo.

> There should be a way to remove those "left-over" files generated from older CLDRs.

This is the "overwrite" option.  There are two choices: exit if the directory is non-empty, or delete the entire directory before continuing.  There is no option to write into an already-populated directory for reasons that you stated.